### PR TITLE
predict doors and airlocks

### LINKED ISSUE #19032: EMAG sparks don't appear anymore
## Description

i swear we used to have this.

### LINKED ISSUE #19382: Door prying is not predicted
Door code in general needs a good once-over but this should be easy idk.

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR

adds prediction to doors and airlocks.

additionally fixes slight animation error

fixes #19382 
fixes #19032 

## Why / Balance

they felt like ass.

## Technical details

moves almost all of door and airlock code into shared. additionally combines doorbolt system and doorsystem.

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

DoorBoltSystem is folded into DoorSystem.

A lot of method calls for doors have been modified to take Entity<T>.

**Changelog**

no cl no fun

### CODE REVIEW

File: Content.Shared/Prying/Systems/PryingSystem.cs [hunk 1]
```diff
@@ -162,23 +162,16 @@ private void OnDoAfter(EntityUid uid, DoorComponent door, DoorPryDoAfterEvent ar
...
         {
-            if (message != null)
-                Popup.PopupEntity(Loc.GetString(message), uid, args.User);
+            if (!string.IsNullOrWhiteSpace(message))
+                Popup.PopupClient(Loc.GetString(message), uid, args.User);
             return;
         }
 
         // TODO: When we get airlock prediction make this fully predicted.
         // When that happens also fix the checking function in the Client AirlockSystem.
```
Comment (nikthechampiongr) [inline Content.Shared/Prying/Systems/PryingSystem.cs:171]: You should probably remove this todo

File: Content.Shared/Doors/Components/DoorComponent.cs [hunk 2]
```diff
@@ -302,6 +302,9 @@ public sealed partial class DoorComponent : Component
 
     [DataField(customTypeSerializer: typeof(ConstantSerializer<DrawDepthTag>))]
     public int ClosedDrawDepth = (int) DrawDepth.DrawDepth.Doors;
+
+    [DataField, AutoNetworkedField]
+    public bool Powered;
 }
```
Comment (metalgearsloth) [inline Content.Shared/Doors/Components/DoorComponent.cs:308]: Doors don't require power, e.g. stone doors are valid. Adding this field to 3 comps is kinda nasty.

**Review by metalgearsloth — APPROVED** [review]
Rest looks okay just the power stuff is cooked.

**Comment by EmoGarbage404** [comment]
> Rest looks okay just the power stuff is cooked.

turns out it wasn't even necessary to have :P

**Pull request merged**